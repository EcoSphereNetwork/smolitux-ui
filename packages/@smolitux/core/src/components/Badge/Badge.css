.smolitux-badge {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  font-size: 0.75rem;
  font-weight: 500;
  line-height: 1;
  padding: 0.25rem 0.5rem;
  border-radius: 0.25rem;
  white-space: nowrap;
  vertical-align: middle;
  color: var(--badge-text-color, #1f2937);
  background-color: var(--badge-bg, #e5e7eb);
}

/* Varianten */
.smolitux-badge--default {
  color: var(--badge-default-text-color, #1f2937);
  background-color: var(--badge-default-bg, #e5e7eb);
}

.smolitux-badge--primary {
  color: var(--badge-primary-text-color, white);
  background-color: var(--badge-primary-bg, #3b82f6);
}

.smolitux-badge--success {
  color: var(--badge-success-text-color, white);
  background-color: var(--badge-success-bg, #10b981);
}

.smolitux-badge--warning {
  color: var(--badge-warning-text-color, #1f2937);
  background-color: var(--badge-warning-bg, #f59e0b);
}

.smolitux-badge--error {
  color: var(--badge-error-text-color, white);
  background-color: var(--badge-error-bg, #ef4444);
}

.smolitux-badge--info {
  color: var(--badge-info-text-color, white);
  background-color: var(--badge-info-bg, #3b82f6);
}

/* Größen */
.smolitux-badge--xs {
  font-size: 0.625rem;
  padding: 0.125rem 0.375rem;
}

.smolitux-badge--sm {
  font-size: 0.75rem;
  padding: 0.25rem 0.5rem;
}

.smolitux-badge--md {
  font-size: 0.875rem;
  padding: 0.25rem 0.625rem;
}

.smolitux-badge--lg {
  font-size: 1rem;
  padding: 0.375rem 0.75rem;
}

/* Abgerundete Variante */
.smolitux-badge--rounded {
  border-radius: 9999px;
}

/* Zähler-Variante */
.smolitux-badge--counter {
  min-width: 1.5rem;
  height: 1.5rem;
  padding: 0 0.375rem;
  border-radius: 9999px;
}

.smolitux-badge--counter.smolitux-badge--xs {
  min-width: 1.25rem;
  height: 1.25rem;
}

.smolitux-badge--counter.smolitux-badge--lg {
  min-width: 1.75rem;
  height: 1.75rem;
}

/* Punkt-Variante */
.smolitux-badge--dot {
  width: 0.5rem;
  height: 0.5rem;
  padding: 0;
  border-radius: 50%;
}

.smolitux-badge--dot.smolitux-badge--xs {
  width: 0.375rem;
  height: 0.375rem;
}

.smolitux-badge--dot.smolitux-badge--sm {
  width: 0.5rem;
  height: 0.5rem;
}

.smolitux-badge--dot.smolitux-badge--md {
  width: 0.625rem;
  height: 0.625rem;
}

.smolitux-badge--dot.smolitux-badge--lg {
  width: 0.75rem;
  height: 0.75rem;
}

/* Outline-Variante */
.smolitux-badge--outline {
  background-color: transparent;
  border: 1px solid;
}

.smolitux-badge--outline.smolitux-badge--default {
  color: var(--badge-default-bg, #9ca3af);
  border-color: var(--badge-default-bg, #9ca3af);
}

.smolitux-badge--outline.smolitux-badge--primary {
  color: var(--badge-primary-bg, #3b82f6);
  border-color: var(--badge-primary-bg, #3b82f6);
}

.smolitux-badge--outline.smolitux-badge--success {
  color: var(--badge-success-bg, #10b981);
  border-color: var(--badge-success-bg, #10b981);
}

.smolitux-badge--outline.smolitux-badge--warning {
  color: var(--badge-warning-bg, #f59e0b);
  border-color: var(--badge-warning-bg, #f59e0b);
}

.smolitux-badge--outline.smolitux-badge--error {
  color: var(--badge-error-bg, #ef4444);
  border-color: var(--badge-error-bg, #ef4444);
}

.smolitux-badge--outline.smolitux-badge--info {
  color: var(--badge-info-bg, #3b82f6);
  border-color: var(--badge-info-bg, #3b82f6);
}

/* Badge mit Icon */
.smolitux-badge-icon {
  display: inline-flex;
  align-items: center;
  margin-right: 0.25rem;
}

/* Animationen */
@keyframes smolitux-badge-pulse {
  0% {
    transform: scale(1);
  }
  50% {
    transform: scale(1.05);
  }
  100% {
    transform: scale(1);
  }
}

.smolitux-badge--animate-pulse {
  animation: smolitux-badge-pulse 2s infinite;
}

/* Dark Mode */
.dark .smolitux-badge--default {
  color: var(--badge-default-text-color-dark, #f9fafb);
  background-color: var(--badge-default-bg-dark, #4b5563);
}

.dark .smolitux-badge--primary {
  color: var(--badge-primary-text-color-dark, white);
  background-color: var(--badge-primary-bg-dark, #2563eb);
}

.dark .smolitux-badge--success {
  color: var(--badge-success-text-color-dark, white);
  background-color: var(--badge-success-bg-dark, #059669);
}

.dark .smolitux-badge--warning {
  color: var(--badge-warning-text-color-dark, #1f2937);
  background-color: var(--badge-warning-bg-dark, #d97706);
}

.dark .smolitux-badge--error {
  color: var(--badge-error-text-color-dark, white);
  background-color: var(--badge-error-bg-dark, #dc2626);
}

.dark .smolitux-badge--info {
  color: var(--badge-info-text-color-dark, white);
  background-color: var(--badge-info-bg-dark, #2563eb);
}

.dark .smolitux-badge--outline.smolitux-badge--default {
  color: var(--badge-default-bg-dark, #9ca3af);
  border-color: var(--badge-default-bg-dark, #9ca3af);
}

.dark .smolitux-badge--outline.smolitux-badge--primary {
  color: var(--badge-primary-bg-dark, #60a5fa);
  border-color: var(--badge-primary-bg-dark, #60a5fa);
}

.dark .smolitux-badge--outline.smolitux-badge--success {
  color: var(--badge-success-bg-dark, #34d399);
  border-color: var(--badge-success-bg-dark, #34d399);
}

.dark .smolitux-badge--outline.smolitux-badge--warning {
  color: var(--badge-warning-bg-dark, #fbbf24);
  border-color: var(--badge-warning-bg-dark, #fbbf24);
}

.dark .smolitux-badge--outline.smolitux-badge--error {
  color: var(--badge-error-bg-dark, #f87171);
  border-color: var(--badge-error-bg-dark, #f87171);
}

.dark .smolitux-badge--outline.smolitux-badge--info {
  color: var(--badge-info-bg-dark, #60a5fa);
  border-color: var(--badge-info-bg-dark, #60a5fa);
}