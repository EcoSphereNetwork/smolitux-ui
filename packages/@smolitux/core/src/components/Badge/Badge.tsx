import React from 'react';

export type BadgeVariant = 'default' | 'primary' | 'success' | 'warning' | 'error' | 'info';
export type BadgeSize = 'xs' | 'sm' | 'md' | 'lg';

export interface BadgeProps {
  /** Anzeigetext */
  children: React.ReactNode;
  /** Variante der Badge */
  variant?: BadgeVariant;
  /** Größe der Badge */
  size?: BadgeSize; 
  /** Abgerundeter Stil */
  rounded?: boolean;
  /** Zusätzliche CSS-Klassen */
  className?: string;
  /** Optional Icon */
  icon?: React.ReactNode;
  /** Ob die Badge als Zähler angezeigt werden soll (z.B. für Benachrichtigungen) */
  isCounter?: boolean;
  /** Maximaler Wert für Zähler (z.B. 99+) */
  maxCount?: number;
  /** Ob die Badge als Punkt ohne Text angezeigt werden soll */
  isDot?: boolean;
  /** Ob die Badge als Outline angezeigt werden soll */
  outline?: boolean;
  /** Zusätzliche HTML-Attribute */
  htmlProps?: React.HTMLAttributes<HTMLSpanElement>;
  /** ID für Barrierefreiheit */
  id?: string;
}

/**
 * Badge-Komponente zum Anzeigen von Status oder Kennzeichnungen
 * 
 * @example
 * ```tsx
 * <Badge>Default</Badge>
 * <Badge variant="primary">Primary</Badge>
 * <Badge variant="success" icon={<CheckIcon />}>Success</Badge>
 * <Badge variant="error" rounded>Error</Badge>
 * <Badge isCounter maxCount={99}>100</Badge>
 * <Badge isDot variant="warning" />
 * ```
 */
export const Badge: React.FC<BadgeProps> = ({
  children,
  variant = 'default',
  size = 'md',
  rounded = false,
  className = '',
  icon,
  isCounter = false,
  maxCount,
  isDot = false,
  outline = false,
  htmlProps = {},
  id
}) => {
  // Varianten-spezifische Klassen
  const variantClasses = {
    default: outline 
      ? 'bg-transparent border border-gray-300 text-gray-800 dark:border-gray-600 dark:text-gray-300' 
      : 'bg-gray-100 text-gray-800 dark:bg-gray-700 dark:text-gray-300',
    primary: outline 
      ? 'bg-transparent border border-primary-500 text-primary-700 dark:border-primary-400 dark:text-primary-300' 
      : 'bg-primary-100 text-primary-800 dark:bg-primary-900/30 dark:text-primary-300',
    success: outline 
      ? 'bg-transparent border border-green-500 text-green-700 dark:border-green-400 dark:text-green-300' 
      : 'bg-green-100 text-green-800 dark:bg-green-900/30 dark:text-green-300',
    warning: outline 
      ? 'bg-transparent border border-yellow-500 text-yellow-700 dark:border-yellow-400 dark:text-yellow-300' 
      : 'bg-yellow-100 text-yellow-800 dark:bg-yellow-900/30 dark:text-yellow-300',
    error: outline 
      ? 'bg-transparent border border-red-500 text-red-700 dark:border-red-400 dark:text-red-300' 
      : 'bg-red-100 text-red-800 dark:bg-red-900/30 dark:text-red-300',
    info: outline 
      ? 'bg-transparent border border-blue-500 text-blue-700 dark:border-blue-400 dark:text-blue-300' 
      : 'bg-blue-100 text-blue-800 dark:bg-blue-900/30 dark:text-blue-300'
  };

  // Größen-spezifische Klassen
  const sizeClasses = {
    xs: isDot ? 'h-1.5 w-1.5' : 'text-xs px-1 py-0.5 min-w-[1rem]',
    sm: isDot ? 'h-2 w-2' : 'text-xs px-1.5 py-0.5 min-w-[1.25rem]',
    md: isDot ? 'h-2.5 w-2.5' : 'text-xs px-2.5 py-0.5 min-w-[1.5rem]',
    lg: isDot ? 'h-3 w-3' : 'text-sm px-3 py-1 min-w-[1.75rem]'
  };

  // Form-Klassen
  const roundedClass = isDot 
    ? 'rounded-full' 
    : (rounded ? 'rounded-full' : 'rounded-md');

  // Zähler-Formatierung
  let displayContent = children;
  if (isCounter && typeof children === 'string' || typeof children === 'number') {
    const count = Number(children);
    if (!isNaN(count) && maxCount && count > maxCount) {
      displayContent = `${maxCount}+`;
    }
  }

  // Kombinierte Klassen
  const badgeClasses = [
    isDot ? '' : 'inline-flex items-center justify-center font-medium',
    variantClasses[variant],
    sizeClasses[size],
    roundedClass,
    isCounter && !isDot ? 'text-center' : '',
    className
  ].filter(Boolean).join(' ');

  // Barrierefreiheits-Attribute
  const ariaProps = {
    role: 'status',
    ...htmlProps,
    ...(id ? { id } : {})
  };

  if (isDot) {
    return (
      <span 
        className={badgeClasses} 
        aria-hidden="true"
        {...ariaProps}
      />
    );
  }

  return (
    <span 
      className={badgeClasses}
      {...ariaProps}
    >
      {icon && <span className="mr-1" aria-hidden="true">{icon}</span>}
      {displayContent}
    </span>
  );
};

export default Badge;
