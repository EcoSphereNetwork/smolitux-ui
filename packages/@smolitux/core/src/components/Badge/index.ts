export { Badge } from './Badge';
export type { BadgeProps } from './Badge';

// Für Abwärtskompatibilität mit dem bestehenden Export
export { Badge as default } from './Badge';