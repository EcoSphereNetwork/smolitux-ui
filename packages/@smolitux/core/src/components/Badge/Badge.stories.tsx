import React from 'react';
import { Meta, StoryObj } from '@storybook/react';
import Badge from './Badge';

const meta: Meta<typeof Badge> = {
  title: 'Core/Badge',
  component: Badge,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  argTypes: {
    variant: {
      control: { type: 'select' },
      options: ['default', 'primary', 'success', 'warning', 'error', 'info'],
    },
    size: {
      control: { type: 'select' },
      options: ['xs', 'sm', 'md', 'lg'],
    },
    rounded: { control: 'boolean' },
    isCounter: { control: 'boolean' },
    maxCount: { control: 'number' },
    isDot: { control: 'boolean' },
    outline: { control: 'boolean' },
    children: { control: 'text' },
  },
};

export default meta;
type Story = StoryObj<typeof Badge>;

export const Default: Story = {
  args: {
    children: 'Badge',
    variant: 'default',
    size: 'md',
  },
};

export const Primary: Story = {
  args: {
    children: 'Primary',
    variant: 'primary',
    size: 'md',
  },
};

export const Success: Story = {
  args: {
    children: 'Success',
    variant: 'success',
    size: 'md',
  },
};

export const Warning: Story = {
  args: {
    children: 'Warning',
    variant: 'warning',
    size: 'md',
  },
};

export const Error: Story = {
  args: {
    children: 'Error',
    variant: 'error',
    size: 'md',
  },
};

export const Info: Story = {
  args: {
    children: 'Info',
    variant: 'info',
    size: 'md',
  },
};

export const Rounded: Story = {
  args: {
    children: 'Rounded',
    variant: 'primary',
    size: 'md',
    rounded: true,
  },
};

export const Outline: Story = {
  args: {
    children: 'Outline',
    variant: 'primary',
    size: 'md',
    outline: true,
  },
};

export const WithIcon: Story = {
  args: {
    children: 'With Icon',
    variant: 'success',
    size: 'md',
    icon: (
      <svg className="w-3 h-3" fill="none" stroke="currentColor" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
        <path strokeLinecap="round" strokeLinejoin="round" strokeWidth="2" d="M5 13l4 4L19 7"></path>
      </svg>
    ),
  },
};

export const Counter: Story = {
  args: {
    children: '42',
    variant: 'primary',
    size: 'md',
    isCounter: true,
  },
};

export const CounterWithMax: Story = {
  args: {
    children: '120',
    variant: 'error',
    size: 'md',
    isCounter: true,
    maxCount: 99,
  },
};

export const Dot: Story = {
  args: {
    variant: 'success',
    size: 'md',
    isDot: true,
  },
};

export const Sizes: Story = {
  render: () => (
    <div className="flex items-center space-x-4">
      <Badge size="xs">XS</Badge>
      <Badge size="sm">SM</Badge>
      <Badge size="md">MD</Badge>
      <Badge size="lg">LG</Badge>
    </div>
  ),
};

export const DotSizes: Story = {
  render: () => (
    <div className="flex items-center space-x-4">
      <Badge size="xs" isDot variant="error" />
      <Badge size="sm" isDot variant="error" />
      <Badge size="md" isDot variant="error" />
      <Badge size="lg" isDot variant="error" />
    </div>
  ),
};

export const AllVariants: Story = {
  render: () => (
    <div className="flex flex-col space-y-4">
      <div className="flex items-center space-x-2">
        <Badge variant="default">Default</Badge>
        <Badge variant="primary">Primary</Badge>
        <Badge variant="success">Success</Badge>
        <Badge variant="warning">Warning</Badge>
        <Badge variant="error">Error</Badge>
        <Badge variant="info">Info</Badge>
      </div>
      <div className="flex items-center space-x-2">
        <Badge variant="default" outline>Default</Badge>
        <Badge variant="primary" outline>Primary</Badge>
        <Badge variant="success" outline>Success</Badge>
        <Badge variant="warning" outline>Warning</Badge>
        <Badge variant="error" outline>Error</Badge>
        <Badge variant="info" outline>Info</Badge>
      </div>
    </div>
  ),
};

export const InContext: Story = {
  render: () => (
    <div className="flex flex-col space-y-4">
      <div className="flex items-center">
        <span className="mr-2">Nachrichten</span>
        <Badge variant="primary" isCounter>5</Badge>
      </div>
      <div className="flex items-center">
        <span className="mr-2">Status:</span>
        <Badge variant="success">Online</Badge>
      </div>
      <div className="flex items-center">
        <span className="mr-2">Benachrichtigungen</span>
        <Badge variant="error" isCounter maxCount={99}>120</Badge>
      </div>
      <div className="flex items-center">
        <span className="mr-2">Neue Funktion</span>
        <Badge variant="info" size="sm">Neu</Badge>
      </div>
    </div>
  ),
};