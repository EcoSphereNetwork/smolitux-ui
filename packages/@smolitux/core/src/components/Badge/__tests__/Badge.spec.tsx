import React from 'react';
import renderer from 'react-test-renderer';
import { Badge } from '../Badge';

describe('Badge Snapshots', () => {
  test('renders correctly with default props', () => {
    const tree = renderer.create(<Badge>Default</Badge>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with primary variant', () => {
    const tree = renderer.create(<Badge variant="primary">Primary</Badge>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with success variant', () => {
    const tree = renderer.create(<Badge variant="success">Success</Badge>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with warning variant', () => {
    const tree = renderer.create(<Badge variant="warning">Warning</Badge>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with error variant', () => {
    const tree = renderer.create(<Badge variant="error">Error</Badge>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with info variant', () => {
    const tree = renderer.create(<Badge variant="info">Info</Badge>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with xs size', () => {
    const tree = renderer.create(<Badge size="xs">XS</Badge>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with sm size', () => {
    const tree = renderer.create(<Badge size="sm">SM</Badge>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with md size', () => {
    const tree = renderer.create(<Badge size="md">MD</Badge>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with lg size', () => {
    const tree = renderer.create(<Badge size="lg">LG</Badge>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with rounded style', () => {
    const tree = renderer.create(<Badge rounded>Rounded</Badge>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with custom className', () => {
    const tree = renderer.create(<Badge className="custom-class">Custom</Badge>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with icon', () => {
    const icon = <span>â˜…</span>;
    const tree = renderer.create(<Badge icon={icon}>With Icon</Badge>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly as counter with maxCount', () => {
    const tree = renderer.create(<Badge isCounter maxCount={99}>100</Badge>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly as dot without text', () => {
    const tree = renderer.create(<Badge isDot variant="warning" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with outline style', () => {
    const tree = renderer.create(<Badge outline>Outline</Badge>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with custom ID', () => {
    const tree = renderer.create(<Badge id="test-badge-id">Custom ID</Badge>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly with multiple props combined', () => {
    const icon = <span>â˜…</span>;
    const tree = renderer.create(
      <Badge 
        variant="primary" 
        size="lg" 
        rounded 
        icon={icon} 
        outline
        className="custom-class"
        id="test-badge"
      >
        Combined
      </Badge>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});