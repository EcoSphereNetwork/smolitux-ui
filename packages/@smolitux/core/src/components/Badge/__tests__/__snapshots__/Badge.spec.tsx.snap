// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge Snapshots renders correctly as counter with maxCount 1`] = `
<span
  className="inline-flex items-center justify-center font-medium bg-gray-100 text-gray-800 dark:bg-gray-700 dark:text-gray-300 text-xs px-2.5 py-0.5 min-w-[1.5rem] rounded-md text-center"
  role="status"
>
  99+
</span>
`;

exports[`Badge Snapshots renders correctly as dot without text 1`] = `
<span
  aria-hidden="true"
  className="bg-yellow-100 text-yellow-800 dark:bg-yellow-900/30 dark:text-yellow-300 h-2.5 w-2.5 rounded-full"
  role="status"
/>
`;

exports[`Badge Snapshots renders correctly with custom ID 1`] = `
<span
  className="inline-flex items-center justify-center font-medium bg-gray-100 text-gray-800 dark:bg-gray-700 dark:text-gray-300 text-xs px-2.5 py-0.5 min-w-[1.5rem] rounded-md"
  id="test-badge-id"
  role="status"
>
  Custom ID
</span>
`;

exports[`Badge Snapshots renders correctly with custom className 1`] = `
<span
  className="inline-flex items-center justify-center font-medium bg-gray-100 text-gray-800 dark:bg-gray-700 dark:text-gray-300 text-xs px-2.5 py-0.5 min-w-[1.5rem] rounded-md custom-class"
  role="status"
>
  Custom
</span>
`;

exports[`Badge Snapshots renders correctly with default props 1`] = `
<span
  className="inline-flex items-center justify-center font-medium bg-gray-100 text-gray-800 dark:bg-gray-700 dark:text-gray-300 text-xs px-2.5 py-0.5 min-w-[1.5rem] rounded-md"
  role="status"
>
  Default
</span>
`;

exports[`Badge Snapshots renders correctly with error variant 1`] = `
<span
  className="inline-flex items-center justify-center font-medium bg-red-100 text-red-800 dark:bg-red-900/30 dark:text-red-300 text-xs px-2.5 py-0.5 min-w-[1.5rem] rounded-md"
  role="status"
>
  Error
</span>
`;

exports[`Badge Snapshots renders correctly with icon 1`] = `
<span
  className="inline-flex items-center justify-center font-medium bg-gray-100 text-gray-800 dark:bg-gray-700 dark:text-gray-300 text-xs px-2.5 py-0.5 min-w-[1.5rem] rounded-md"
  role="status"
>
  <span
    aria-hidden="true"
    className="mr-1"
  >
    <span>
      ★
    </span>
  </span>
  With Icon
</span>
`;

exports[`Badge Snapshots renders correctly with info variant 1`] = `
<span
  className="inline-flex items-center justify-center font-medium bg-blue-100 text-blue-800 dark:bg-blue-900/30 dark:text-blue-300 text-xs px-2.5 py-0.5 min-w-[1.5rem] rounded-md"
  role="status"
>
  Info
</span>
`;

exports[`Badge Snapshots renders correctly with lg size 1`] = `
<span
  className="inline-flex items-center justify-center font-medium bg-gray-100 text-gray-800 dark:bg-gray-700 dark:text-gray-300 text-sm px-3 py-1 min-w-[1.75rem] rounded-md"
  role="status"
>
  LG
</span>
`;

exports[`Badge Snapshots renders correctly with md size 1`] = `
<span
  className="inline-flex items-center justify-center font-medium bg-gray-100 text-gray-800 dark:bg-gray-700 dark:text-gray-300 text-xs px-2.5 py-0.5 min-w-[1.5rem] rounded-md"
  role="status"
>
  MD
</span>
`;

exports[`Badge Snapshots renders correctly with multiple props combined 1`] = `
<span
  className="inline-flex items-center justify-center font-medium bg-transparent border border-primary-500 text-primary-700 dark:border-primary-400 dark:text-primary-300 text-sm px-3 py-1 min-w-[1.75rem] rounded-full custom-class"
  id="test-badge"
  role="status"
>
  <span
    aria-hidden="true"
    className="mr-1"
  >
    <span>
      ★
    </span>
  </span>
  Combined
</span>
`;

exports[`Badge Snapshots renders correctly with outline style 1`] = `
<span
  className="inline-flex items-center justify-center font-medium bg-transparent border border-gray-300 text-gray-800 dark:border-gray-600 dark:text-gray-300 text-xs px-2.5 py-0.5 min-w-[1.5rem] rounded-md"
  role="status"
>
  Outline
</span>
`;

exports[`Badge Snapshots renders correctly with primary variant 1`] = `
<span
  className="inline-flex items-center justify-center font-medium bg-primary-100 text-primary-800 dark:bg-primary-900/30 dark:text-primary-300 text-xs px-2.5 py-0.5 min-w-[1.5rem] rounded-md"
  role="status"
>
  Primary
</span>
`;

exports[`Badge Snapshots renders correctly with rounded style 1`] = `
<span
  className="inline-flex items-center justify-center font-medium bg-gray-100 text-gray-800 dark:bg-gray-700 dark:text-gray-300 text-xs px-2.5 py-0.5 min-w-[1.5rem] rounded-full"
  role="status"
>
  Rounded
</span>
`;

exports[`Badge Snapshots renders correctly with sm size 1`] = `
<span
  className="inline-flex items-center justify-center font-medium bg-gray-100 text-gray-800 dark:bg-gray-700 dark:text-gray-300 text-xs px-1.5 py-0.5 min-w-[1.25rem] rounded-md"
  role="status"
>
  SM
</span>
`;

exports[`Badge Snapshots renders correctly with success variant 1`] = `
<span
  className="inline-flex items-center justify-center font-medium bg-green-100 text-green-800 dark:bg-green-900/30 dark:text-green-300 text-xs px-2.5 py-0.5 min-w-[1.5rem] rounded-md"
  role="status"
>
  Success
</span>
`;

exports[`Badge Snapshots renders correctly with warning variant 1`] = `
<span
  className="inline-flex items-center justify-center font-medium bg-yellow-100 text-yellow-800 dark:bg-yellow-900/30 dark:text-yellow-300 text-xs px-2.5 py-0.5 min-w-[1.5rem] rounded-md"
  role="status"
>
  Warning
</span>
`;

exports[`Badge Snapshots renders correctly with xs size 1`] = `
<span
  className="inline-flex items-center justify-center font-medium bg-gray-100 text-gray-800 dark:bg-gray-700 dark:text-gray-300 text-xs px-1 py-0.5 min-w-[1rem] rounded-md"
  role="status"
>
  XS
</span>
`;
